nengo_viz/static/viz_slider.js

```diff
@@ -212,12 +226,105 @@ VIZ.Slider.prototype.generate_menu = function() {
+    var self = this;
+    if (self.filling_slider_value){
+        return;
+    }
+    else{
+        VIZ.netgraph.menu.hide_any();
+        self.filling_slider_value = true;
+        var text_div = this.sliders[ind].value_display;
+        var original_value = text_div.innerHTML;
+        text_div.innerHTML = '<input id="value_in_field" style=" border:0; outline:0;"></input>';
+        elem = document.getElementById('value_in_field')
+        elem.value = Number(original_value);
+        elem.focus();
+        elem.select();
+        elem.style.width = '3em';
+        elem.style.textAlign = 'center';
+        $(text_div).on('keypress', function (event) {self.submit_value(event.which, ind, text_div);});
+    }
+};
+
+VIZ.Slider.prototype.submit_value = function (button, ind, text_div) {
+    if (button == 13) {
+        var self = this;
+        var slider_range = self.scale.domain();
+        var msg = document.getElementById('value_in_field').value;
+        $(text_div).off('keypress');
+        if (VIZ.is_num(msg)) {
+            self.set_value(ind, VIZ.max_min(Number(msg), slider_range[1], slider_range[0]));
+            return;
+        }
+        else {
+            alert('failed to set value');
+            text_div.innerHTML = 0;
+            self.set_value(ind, VIZ.max_min(0, slider_range[1], slider_range[0]));   
+            return;
+        }
+    }
+};
+
+/** report an event back to the server */
+VIZ.Slider.prototype.notify = function(info) {
+    this.notify_msgs.push(info);
+    
+    // only send one message at a time
+    // TODO: find a better way to figure out when it's safe to send
+    // another message, rather than just waiting 1ms....
+    if (this.notify_msgs.length == 1) {
+        var self = this;
+        window.setTimeout(function() {
+            self.send_notify_msg();
+        }, 1);
+    }
+}
+
+/** send exactly one message back to server
+ *  and schedule the next message to be sent, if any
+ */
+VIZ.Slider.prototype.send_notify_msg = function() {
+    msg = this.notify_msgs[0];
+    this.ws.send(msg);
+    if (this.notify_msgs.length > 1) {
+        var self = this;
+        window.setTimeout(function() {
+            self.send_notify_msg();
+        }, 1);
+    }
+    this.notify_msgs.splice(0, 1);
+}
+
+VIZ.Slider.prototype.user_value = function () {
+    var self = this;
+    var new_value = prompt('Set value\nExample: 1.3, 4.2');
+    
+    if (new_value == null) {
+        return;
+    };
+    new_value = new_value.split(',');
+    var slider_range = this.scale.domain();
+
+    for (var i = 0; i < this.sliders.length; i++){
+        if (!(VIZ.is_num(new_value[i]))) {
+            alert("invalid input :" + new_value[i] + "\nFor the slider in position " + (i +1) );
+            break;
+        }
+        insert_value = VIZ.max_min(new_value[i], slider_range[1], slider_range[0]);
+
+        self.set_value(i, insert_value);
+    }
+    //this.save_layout();
```

Comment:
never mind; I see you removed it in a separate PR.  Note that this is why we try to keep PRs self-contained :)